### PR TITLE
Switch to JDK8

### PR DESCRIPTION
**Pull request by damianszczepanik**
- Add tests for FactsBuilder
 - Skip facts with zero value

### CODE REVIEW

File: src/main/java/jenkins/plugins/office365connector/FactsBuilder.java [hunk 1]
```diff
@@ -109,10 +109,7 @@ public void addCulprits(Set<User> authors) {
             return;
         }
 
-        Set<String> culprits = new HashSet<>();
-        for (User user : authors) {
-            culprits.add(user.getFullName());
-        }
+        List<String> culprits = authors.stream().map(user -> user.getFullName()).collect(Collectors.toList());
```
Comment (casz) [inline src/main/java/jenkins/plugins/office365connector/FactsBuilder.java]: Use method reference
```suggestion
        List<String> culprits = authors.stream().map(User::getFullName).collect(Collectors.toList());
```

**Comment by casz** [comment]
I implore you to disable codacy-bot comments and just modify the PR quality review status. Configurable at: https://app.codacy.com/app/YOUR-PROJECT-URL/settings/quality 
Super noisy and all the deleted comment events don't help.

**Comment by damianszczepanik** [comment]
@casz  JDK8 because end of support for jdk7
btw, did you update plugin to the recent release?

**Comment by casz** [comment]
I no longer use Microsoft teams :sweat_smile:

**Pull request merged**